### ISSUE TITLE
Efficient iter_slices for LazyFrames

### ISSUE THREAD

**Issue by chriss1245** [pola-rs/polars#13252]
### Description

I am working in a project using a big dataset with tensorflow. I can not fit in memory all the data at once so I need to constantly retrieve it from disk to pass it to the neural net.. I am wondering whether there is a way to speed up this process of loading batches for the model. Currently I was trying with LazyFrame.slice. But it is too slow. Is it possible to create a more efficient iter_slices?

For instance pyarrow.ParquetFile has iter_batches.

**Comment by deanm0000** [comment]
Your issue is too vague to be actionable.

Are you doing something like ```scan_parquet().slice(n).collect()``` or do you have some transformations too?

**Comment by chriss1245** [comment]
> Your issue is too vague to be actionable.
> 
> Are you doing something like ```scan_parquet().slice(n).collect()``` or do you have some transformations too?

Yes I was trying with that. I take the length of the dataset and I call a slice at a time in a loop.

```

ldf = scan_parquet()
 ...

for i in range (0, ldf_size, batch_size):

    df = ldf.slice(i, batch_size).collect()
    ...

```

 Then I convert everything to a dictionary of tensorflow arrays. I process what I need using tensorflow.
